components/eamxx/src/physics/nudging/eamxx_nudging_process_interface.cpp

```diff
@@ -79,10 +115,16 @@ void Nudging::set_grids(const std::shared_ptr<const GridsManager> grids_manager)
   /* ----------------------- WARNING --------------------------------*/
 
   //Now need to read in the file
-  if (m_src_pres_type == TIME_DEPENDENT_3D_PROFILE) {
-    m_num_src_levs = scorpio::get_dimlen(m_datafiles[0],"lev");
-  } else {
+  if (m_src_pres_type == STATIC_1D_VERTICAL_PROFILE) {
     m_num_src_levs = scorpio::get_dimlen(m_static_vertical_pressure_file,"lev");
+  } else {
+    m_num_src_levs = scorpio::get_dimlen(m_datafiles[0],"lev");
```

Comment:
Is it worth checking that all files have the same "lev" dimension?

Reply:
I feel just checking the first file should be enough - I would expect it's a consensus for nudging users that nudging data should be prepared in the same format because they should come from the same set of source data (I can't currently imagine a situation where different formats of nudging data are used interchangeably). Moreover, the TIME_DEPENDENT_3D_HYBRID condition is even stricter (hybrid coordinate same as the model coordinate)

Reply:
What if the user accidentally uses a wrong pattern for, say, the 2nd file? The run will start, proceed for a while, and then crash. If a restart file was written sufficiently close, that may not be _too_ bad, but it would be certainly more convenient to find this out early during init time, no? And btw, you could probably do this check on the file inside the Nudging constructor, so that the error happens even earlier in the init sequence.

Reply:
thanks! please see if the changes make sense